code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/OptimizeExtension.java

```diff
@@ -868,7 +871,8 @@ Map<DecisionScope, OptimizeProposition> getPropositionsInProgress() {
     @VisibleForTesting
     void setPropositionsInProgress(
             final Map<DecisionScope, OptimizeProposition> propositionsInProgress) {
-        this.propositionsInProgress = propositionsInProgress;
+        OptimizeExtension.propositionsInProgress.clear();
```

Comment:
@spoorthipujariadobe Did this change because after declaring the map as static and final we can't update the reference because of which `this.propositionsInProgress = propositionsInProgress;` was giving error, and since this method was only used by testing class i've used `clear` and `pullAll` methods rather updating the reference